test/postControllerTest.js

```diff
@@ -0,0 +1,160 @@
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+const assert = require('assert');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+
+describe('postController', () => {
+  beforeEach(() => {
+    sinon.stub(postService, 'getAllPosts').resolves([
+      { title: 'post1', author: 'author1', body: 'body1' },
+      { title: 'post2', author: 'author2', body: 'body2' },
+      { title: 'post3', author: 'author3', body: 'body3' },
+    ]);
+    sinon.stub(postService, 'getOnePost').resolves({ title: 'post1', author: 'author1', body: 'body1' });
+    sinon.stub(postService, 'createNewPost').resolves('expected');
+  });
+
+  afterEach(() => {
+    postService.getAllPosts.restore();
+    postService.getOnePost.restore();
+    postService.createNewPost.restore();
+  });
+
+  describe('GET /', () => {
+    it('should call method "postService.getAllPosts" with no arguments', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          sinon.assert.calledWith(postService.getAllPosts);
+        })
+    ));
+    it('should return correct "3 posts" in view', () => (
```

Comment:
I just realized I've been unnecessarily using the ```.then()``` method in a lot of tests where I don't need the stubbed promise to resolve. However why does this specific test also pass? Wouldn't I need to use a ```.then()``` method here as res.render only executes once the stub is resolved?

Reply:
The promises that supertest provides are different than the promises you use in the postController. I believe you don't need to use `.then()` at all in this file (use `.expect()` instead).